src/fmgc/src/guidance/vnav/descent/TodGuidance.ts

```diff
@@ -0,0 +1,107 @@
+        SimVar.SetSimVarValue('K:PAUSE_SET', 'number', 1);
+        let popup = new PopUp();
+        popup.showInformation(title, message, 'small',
+            () => {
+                SimVar.SetSimVarValue('K:PAUSE_SET', 'number', 0);
+                this.cooldown = TIMEOUT;
+                popup = null;
+            });
+    }
+
+    update(deltaTime: number) {
+        this.updateTdReached(deltaTime);
+        this.updateTdPause(deltaTime);
+    }
+
+    updateTdPause(deltaTime: number) {
+        if (
+            this.cooldown <= 0
+            && NXDataStore.get('PAUSE_AT_TOD', 'DISABLED') === 'ENABLED'
+        ) {
+            // Only watching if T/D pause untriggered + between flight phase CLB and CRZ
+            if (!this.tdPaused
+                && this.observer.get().flightPhase >= FmgcFlightPhase.Climb
+                && this.observer.get().flightPhase <= FmgcFlightPhase.Cruise
+                && Simplane.getAutoPilotAirspeedManaged()
+            ) {
+                // Check T/D pause first, then AP mode reversion
+                if ((this.aircraftToDescentProfileRelation.distanceToTopOfDescent() ?? Number.POSITIVE_INFINITY) < parseFloat(NXDataStore.get('PAUSE_AT_TOD_DISTANCE', '10'))) {
+                    this.tdPaused = true;
+                    this.showPausePopup(
+                        'TOP OF DESCENT',
+                        `Paused before the calculated top of descent. System Time was ${new Date().toLocaleTimeString()}.`,
+                    );
+                // Only guard AP above transitional altitude
+                } else if (this.atmosphericConditions.currentAltitude ? this.atmosphericConditions.currentAltitude > this.observer.get().originTransitionAltitude : false) {
+                    const apActive = SimVar.GetSimVarValue('L:A32NX_AUTOPILOT_ACTIVE', 'boolean') && SimVar.GetSimVarValue('L:A32NX_FMA_LATERAL_MODE', 'Enum') === LateralMode.NAV;
+
+                    if (this.apEngaged && !apActive) {
+                        this.showPausePopup(
+                            'AP PROTECTION',
+                            `Autopilot or lateral guindace disengaged before the calculated top of descent. System Time was ${new Date().toLocaleTimeString()}.`,
+                        );
+                    }
+
+                    if (this.apEngaged !== apActive) {
+                        this.apEngaged = apActive;
+                    }
+                }
+            }
+
+            // Reset flags on turnaround
+            if (this.observer.get().flightPhase === FmgcFlightPhase.Done || this.observer.get().flightPhase === FmgcFlightPhase.Preflight) {
+                this.tdPaused = false;
+                this.apEngaged = false;
+            }
+        }
+
+        if (this.cooldown > 0) {
+            this.cooldown = Math.max(0, this.cooldown - deltaTime);
+        }
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    updateTdReached(deltaTime: number) {
```

Comment:
I am not a fan of disabling eslint rules like this. Preferably, we just remove `deltaTime` as an argument to this function or prefix it with an `_` as `_deltaTime` to make the linter happy.